alchemiscale/interface/api.py

```diff
@@ -510,6 +514,57 @@ def cancel_tasks(
     return [str(sk) if sk is not None else None for sk in canceled_sks]
 
 
+@router.post("/bulk/tasks/status/get")
+def tasks_status_get(
+    *,
+    tasks: List[ScopedKey] = Body(embed=True),
+    n4js: Neo4jStore = Depends(get_n4js_depends),
+    token: TokenData = Depends(get_token_data_depends),
+) -> List[Union[str, None]]:
+    valid_tasks = []
+    for task_sk in tasks:
+        try:
+            validate_scopes(task_sk.scope, token)
+            valid_tasks.append(task_sk)
+        except HTTPException:
+            valid_tasks.append(None)
+
+    statuses = n4js.get_task_status(valid_tasks)
+
+    return [status.value if status is not None else None for status in statuses]
+
+
+@router.post("/bulk/tasks/status/set")
+def tasks_status_set(
+    *,
+    tasks: List[ScopedKey] = Body(),
+    status: Optional[str] = Body(),
+    n4js: Neo4jStore = Depends(get_n4js_depends),
+    token: TokenData = Depends(get_token_data_depends),
+) -> List[Union[str, None]]:
+    status = TaskStatusEnum(status)
+    if status not in (
+        TaskStatusEnum.waiting,
+        TaskStatusEnum.invalid,
+        TaskStatusEnum.deleted,
+    ):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"Cannot set status to '{status}', must be one of 'waiting', 'invalid', 'deleted'",
+        )
+
+    tasks_updated = []
+    for task_sk in tasks:
+        try:
+            validate_scopes(task_sk.scope, token)
+        except HTTPException:
+            tasks_updated.append(None)
+        else:
+            tasks_updated.extend(n4js.set_task_status([task_sk], status))
```

Comment:
Correct; we can optimize this further, but I think I'm running out of time on this one. Since we have more complex queries to deal with for status setting, I'd like to make that a future PR.

Reply:
Perfect! raise an issue, and happy to move on.

Reply:
I may have managed to get this one in. :grin: